### PR TITLE
✨ Back up classic disk data in VM ExtraConfig

### PR DESCRIPTION
**Pull request by dilyar85**
**What does this PR do, and why is it needed?**

This PR adds VM classic (static) disk data in a separate EC key when incremental FSS is enabled. This will be used in the auto-registration code to accurately detect partial disk restores from a deleted PVC. In other words, if a changed disk is neither a PVC nor in this class of disk data, it will be considered a restored disk in the auto-registration workflow.

**Which issue(s) is/are addressed by this PR?**

Fixes N/A.

**Are there any special notes for your reviewer**:

Testing Done:
- [x] Deployed a VM with multiple classic disks and verified the expected EC backup in an internal testbed:

```console
$ kubectl get vm -n sdiliyaer-test test-vm-publish -o json | jq .status.volumes
[
  {
    "attached": true,
    "diskUUID": "6000C292-3da9-3f72-f45d-98fc088bdc6a",
    "limit": "2Gi",
    "name": "test-vm-publish_2",
    "type": "Classic",
    "used": "0"
  },
  {
    "attached": true,
    "diskUUID": "6000C298-ea69-2a64-07a7-cd64fd6e8a8d",
    "limit": "10Gi",
    "name": "test-vm-publish",
    "type": "Classic",
    "used": "1Gi"
  },
  {
    "attached": true,
    "diskUUID": "6000C299-27bc-e00d-e82a-9fde79713bfc",
    "limit": "1Gi",
    "name": "test-vm-publish_1",
    "type": "Classic",
    "used": "0"
  }
]

$ govc vm.info -e 'test-vm-publish' | grep vmservice.virtualmachine.classicDiskData
    vmservice.virtualmachine.classicDiskData:            H4sIAAAAAAAA/4quVnLLzEn1S8xNVbJSii7OSCxKTQnLTSvWNYhVKEktLtEty9UtKE3KySzO0Efj65XlpmQr1epQYES8ITUMMYIaEgsAAAD//wEAAP//Ak+M084AAAA=

$ echo "H4sIAAAAAAAA/4quVnLLzEn1S8xNVbJSii7OSCxKTQnLTSvWNYhVKEktLtEty9UtKE3KySzO0Efj65XlpmQr1epQYES8ITUMMYIaEgsAAAD//wEAAP//Ak+M084AAAA=" | base64 -d | gunzip | jq .
[
  {
    "FileName": "[sharedVmfs-0] test-vm-publish/test-vm-publish.vmdk"
  },
  {
    "FileName": "[sharedVmfs-0] test-vm-publish/test-vm-publish_1.vmdk"
  },
  {
    "FileName": "[sharedVmfs-0] test-vm-publish/test-vm-publish_2.vmdk"
  }
]
```

- [x] Add unit tests in  `pkg/providers/vsphere/vmprovider_vm_utils_test.go` for the new function to get attached classic disk UUIDs.
- [x] Add unit tests in `pkg/providers/vsphere/virtualmachine/backup_test.go` for the new backup classic disk data in VM EC.

**Please add a release note if necessary**:

```release-note
Back up classic disk data in VM ExtraConfig to support partial disk restore.
```

### CODE REVIEW

File: pkg/providers/vsphere/virtualmachine/backup.go [hunk 1]
```diff
@@ -46,6 +47,12 @@ type PVCDiskData struct {
 	AccessModes []corev1.PersistentVolumeAccessMode
 }
 
+// ClassicDiskData contains the data of a classic (static) disk attached to VM.
+type ClassicDiskData struct {
```
Comment (aruneshpa) [inline pkg/providers/vsphere/virtualmachine/backup.go:51]: I think we talked about moving this, and the PVCData to the types, so the restore workflow can use it?

Also, move all the backup/restore related fields to a new files in the api/ repo if you want.

**Review by aruneshpa — APPROVED** [review]
LGTM.

File: pkg/providers/vsphere/virtualmachine/backup.go (hunk 1 above)
Comment (dilyar85) [reply pkg/providers/vsphere/virtualmachine/backup.go]: Thanks! Updated as suggested in the latest commit.

File: api/v1alpha3/backup/backup.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
```
Comment (aruneshpa) [inline api/v1alpha3/backup/backup.go:1]: Hmm.  I meant a different "file" not a pacakge.  And if we are considering a different package, should we move it out of the API schema to make sure that it is independent of the schema versions?

**Review by dougm — APPROVED** [review]

**Pull request merged**